### PR TITLE
Namespace and Send/Receive tests

### PR DESCRIPTION
**Pull request by ramya-rao-a**
Apart from the tests, this PR covers
- Adding specific error messages for missing names in create queue/topic/subscription client in namespace.ts, because otherwise you get a "entityPath" related error from the clientEntityContext which is too generic. `entityPath` is not the name of the parameter used when creating queue/topic/subscription clients

The sendRecieve tests cover only the "complete" disposition case. Others will be added later on.

### CODE REVIEW

File: test/namespace.spec.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+describe("Create Namespace", function() {
+  describe(".fromConnectionString", function() {
+    it("throws when there is no connection string", function() {
+      testFalsyValues(function(value: any) {
+        const test = function() {
+          return Namespace.createFromConnectionString(value);
+        };
+        test.should.throw(
+          Error,
+          "'connectionString' is a required parameter and must be of type: 'string'."
```
Comment (AlexGhiondea) [inline test/namespace.spec.ts]: Is this the expected message that it should throw?

File: test/namespace.spec.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+import { Namespace } from "../lib";
+
+function testFalsyValues(testFn: Function) {
+  [null, undefined, "", 0].forEach(function(value) {
+    testFn(value);
+  });
+}
+
+describe("Create Namespace", function() {
+  describe(".fromConnectionString", function() {
```
Comment (AlexGhiondea) [inline test/namespace.spec.ts]: Why do we have 2 nested describes?

File: test/namespace.spec.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+import "mocha";
+import * as chai from "chai";
+const should = chai.should();
+import * as chaiAsPromised from "chai-as-promised";
+import * as dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+import { Namespace } from "../lib";
+
+function testFalsyValues(testFn: Function) {
```
Comment (AlexGhiondea) [inline test/namespace.spec.ts]: Should this be called `testInvalidValues` or something that is more descriptive?

File: test/namespace.spec.ts [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  });
+
+  it("throws when sending batch data via a queueClient to a non existing namespace", function() {
+    const client = namespace.createQueueClient("some-name");
+    const sendPromise = client.sendBatch([{ body: "hello" }]).catch(testError);
+
+    return sendPromise.then(() => should.equal(errorWasThrown, true));
+  });
+
+  it("throws when sending batch data via a topicClient to a non existing namespace", function() {
```
Comment (AlexGhiondea) [inline test/namespace.spec.ts]: Should we split these up based on what they are testing? Queue/Topic/etc?

File: test/sendReceive.spec.ts [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    if (peekedSubscriptionMsg.length) {
+      throw new Error("Please use an empty Subscription for integration testing");
+    }
+  });
+
+  afterEach(() => {
+    return namespace.close();
+  });
+
+  it("Simple send and recieveBatch using Queues", async function() {
```
Comment (AlexGhiondea) [inline test/sendReceive.spec.ts]: receiveBatch

File: test/sendReceive.spec.ts [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  SubscriptionClient
+} from "../lib";
+
+describe("Simple send/receive to/from Queue/Topic/Subscription", function() {
+  let namespace: Namespace;
+  let queueClient: QueueClient;
+  let topicClient: TopicClient;
+  let subscriptionClient: SubscriptionClient;
+
+  beforeEach(async () => {
```
Comment (AlexGhiondea) [inline test/sendReceive.spec.ts]: Is there anything special about the queues? As they need to be created prior to the tests running if they need any special setup we should capture it here.

File: test/namespace.spec.ts (hunk 4 above)
Comment (AlexGhiondea) [reply test/namespace.spec.ts]: Sounds good!

File: test/namespace.spec.ts (hunk 2 above)
Comment (ramya-rao-a) [reply test/namespace.spec.ts]: Fixed

File: test/sendReceive.spec.ts (hunk 6 above)
Comment (ramya-rao-a) [reply test/sendReceive.spec.ts]: Note added

**Pull request merged**